app/src/androidTest/java/com/urbandroid/sleep/addon/stats/model/socialjetlag/SocialJetlagStatsTest.java

```diff
@@ -71,4 +76,169 @@ public void testGetRecordIrregularity() {
+
+        // pretty regular
+        BitSet interval9 = new BitSet(1440);
+        BitSet interval10 = new BitSet(1440);
+        interval9.set(0, 720);
+        interval10.set(100, 820);
+        Assert.assertEquals(1.0 - 200.0/1440.0, stats.calculateSRI(interval9, interval10), .000001);
+    }
+
+    @Test
+    public void testCalculateAvgSRIBase() {
+        ChronoRecords halfDayDiff = new ChronoRecords(Arrays.asList(new ChronoRecord[]{
+                //The first 3 records are in UTC timezone
+                new ChronoRecord(
+                        Date.from(Instant.parse("2018-11-11T01:30:10Z")),
+                        Date.from(Instant.parse("2018-11-11T10:00:10Z")),
+                        1.5f, 10f, 7f),
+                new ChronoRecord(
+                        Date.from(Instant.parse("2018-11-12T01:00:10Z")),
+                        Date.from(Instant.parse("2018-11-12T10:00:10Z")),
+                        1.0f, 10f, 7f),}));
+        SocialJetlagStats  stats = new SocialJetlagStats(halfDayDiff, false);
+        List<Interval> recordsAsIntervals = stats.convertChronoRecordsToTimeIntervals(halfDayDiff);
+        TreeMap<String, BitSet> dayBitMap = stats.createSleepStateByDateMap(recordsAsIntervals);
+        float avgSRI = stats.calculateAverageSRI(dayBitMap);
+        Assert.assertEquals(1.0-30.0/1440.0, avgSRI, 0.001);
+    }
+
+    @Test
+    public void overLappingDays() {
+        ChronoRecords halfDayDiff = new ChronoRecords(Arrays.asList(new ChronoRecord[]{
+                //The first 3 records are in UTC timezone
+                new ChronoRecord(
+                        Date.from(Instant.parse("2018-11-11T00:00:10Z")),
+                        Date.from(Instant.parse("2018-11-11T06:00:10Z")),
+                        10f, 6f, 8f),
+                new ChronoRecord(
+                        Date.from(Instant.parse("2018-11-11T10:00:10Z")),
+                        Date.from(Instant.parse("2018-11-12T06:00:10Z")),
+                        10f, 6f, 8f),
+                new ChronoRecord(
+                        Date.from(Instant.parse("2018-11-12T10:30:10Z")),
+                        Date.from(Instant.parse("2018-11-13T06:30:10Z")),
+                        10.5f, 6.5f, 8f),
+                new ChronoRecord(
+                        Date.from(Instant.parse("2018-11-13T10:00:10Z")),
+                        Date.from(Instant.parse("2018-11-13T23:59:10Z")),
+                        10f, 6f, 8f)}));
+        SocialJetlagStats  stats = new SocialJetlagStats(halfDayDiff, false);
+        List<Interval> recordsAsIntervals = stats.convertChronoRecordsToTimeIntervals(halfDayDiff);
+        TreeMap<String, BitSet> dayBitMap = stats.createSleepStateByDateMap(recordsAsIntervals);
+        float avgSRI = stats.calculateAverageSRI(dayBitMap);
+        Assert.assertEquals(1.0-45.0/1440.0, avgSRI, 0.001);
+    }
+
+    @Test
+    public void testSkipDays() {
```

Comment:
I thought we threw an error / print statement if we skip days - did we not?

Reply:
Ah that's right. It currently just prints, should we have it throw an error instead/as well?